niworkflows/interfaces/mni.py

```diff
@@ -161,42 +197,129 @@ def _get_ants_args(self):
                         self.inputs.reference_image,
-                        self.inputs.mreference_mask,
+                        self.inputs.reference_mask,
                         "fixed_masked.nii.gz")
+
+                    # If a lesion mask is also provided...
+                    if isdefined(self.inputs.lesion_mask):
+                        # Create a cost function mask with the form: [global mask]
+                        # Use this as the fixed mask.
+                        args['fixed_image_mask'] = create_cfm(
+                            self.inputs.reference_mask,
+                            lesion_mask=None,
+                            global_mask=True)
+
+                # If a reference mask is provided...
+                # But explicit masking is disabled...
                 else:
+                    # Use the reference mask as the fixed mask during registration.
+                    # Do not mask the fixed image.
                     args['fixed_image_mask'] = self.inputs.reference_mask
+
+            # If no reference mask is provided...
+            # But a lesion mask *IS* provided ...
+            elif isdefined(self.inputs.lesion_mask):
+                # Create a cost function mask with the form: [global mask]
+                # Use this as the fixed mask
+                args['fixed_image_mask'] = create_cfm(
+                    self.inputs.reference_image,
+                    lesion_mask=None,
+                    global_mask=True)
+
+        # If no reference image is provided, fall back to the default template.
         else:
+            # Raise an error if the user specifies an unsupported image orientation.
             if self.inputs.orientation == 'LAS':
                 raise NotImplementedError
 
+            # Get the template specified by the user.
             mni_template = getters.get_dataset(self.inputs.template)
+            # Set the template resolution.
             resolution = self.inputs.template_resolution
 
+            # If explicit masking is enabled...
             if self.inputs.explicit_masking:
-                args['fixed_image'] = mask(op.join(
-                    mni_template, '%dmm_%s.nii.gz' % (resolution, self.inputs.reference)),
-                    op.join(
-                        mni_template, '%dmm_brainmask.nii.gz' % resolution),
-                    "fixed_masked.nii.gz")
+                # Mask the template image with the template mask.
+                # Do not use a fixed mask during registration.
+                _template_fmt = op.join(mni_template, '%dmm_%s.nii.gz')
+                args['fixed_image'] = mask(
+                        _template_fmt % (resolution, self.inputs.reference),
+                        _template_fmt % (resolution, 'brainmask.nii.gz'),
+                        "fixed_masked.nii.gz")
+
+                # If a lesion mask is provided...
+                if isdefined(self.inputs.lesion_mask):
```

Comment:
After we get the previous comments more clear, I'd dive into this.

Reply:
To be clear, are you talking here about adding a comment for a truth table? I'd be inclined to say that for right now this is a "nice to have" and that it might be worth doing a separate PR for the comments (so we can get the code merged today).